content/2017/03/30/safe-get.rst

```diff
@@ -0,0 +1,257 @@
+.. _first-class functions: https://medium.com/@kaelig/sass-first-class-functions-6e718e2b5eb0
+
+
+Calling Functions
+-----------------
+
+Normally,
+when we're using functions in Sass,
+we know what function we need,
+and we can reference it directly:
+
+.. code:: scss
+
+  // Using Susy's "span" function directly, with a single argument
+  .span {
+    width: span(3);
+  }
+
+But when we build toolkits in Sass,
+it's common that we don't know for sure
+what function we'll be calling.
+In OddBird's `Accoutrement`_ tools
+we even let the user pass in arbitrary functions
+and arguments
+that we'll call at the right time
+to manipulate CSS colors and sizes.
+
+In order to call functions,
+without knowing the function name in advance,
+we have to use the ``call()`` function.
+Here's how it works
+on the current versions of Sass:
+
+.. code:: scss
+
+  // This could change!
+  $function: 'span';
+
+  // Calling some unknown function, with a single argument
+  .span {
+    width: call($function, 3);
+  }
+
+Those two code samples will return the same results.
+The first is more direct,
+but the second is more flexible
+for use in a toolkit.
+
+.. _Accoutrement: /2017/03/07/pattern-making/
+
+
+Introducing the ``get-fuction`` function
+----------------------------------------
+
+Sass is taking a first step towards
+modular name-spacing –
+expected to land in the ``4.0`` release.
+This will allow you to include third-party tools
+without any concern for naming conflicts.
+
+Functions will be name-spaced locally
+to a given Sass file –
+something like ``susy.span()``,
+though the syntax hasn't been settled.
+The new
+``get-function()`` allows us to capture
+a snapshot of a function into a variable,
+pass that snapshot into new name-spaing contexts.
+
+.. code:: scss
+
+  // type-of($my-function) == 'function'
+  $my-function: get-function('susy.span');
+
+In Sass 3.5 and later,
+the ``call()`` function only accepts
+first-class functions,
+where it used to accept function-names as a string.
+In brief,
+we have to start useing
+``get-function('function-name')``
+before calling a function using
+``call()`` –
+but only in new versions of Sass.
+
+.. code:: scss
+
+  $call: call(get-function('susy.span'), $arguments...);
+
+This creates a problem for toolkits and frameworks
+(like `Susy`_)
+that use ``call()`` internally
+to handle user input.
+How do we support old and new versions of Sass,
+while allowing users to pass in
+either strings or fisrt-class functions?
+
+`Kaelig provides one solution`_
+in a great article with more details.
+It's a good start,
+but it doesn't cover all the uses-cases I need.
+What if users pass in a first-class function
+that they've already captured –
+as they likely should in Sass ``3.5+``?
+Here's my slightly-expanded solution.
+
+.. _Susy: http://susy.oddbird.net
+.. _Kaelig provides one solution: https://medium.com/@kaelig/sass-first-class-functions-6e718e2b5eb0
+
+
+Safe Get-Function
```

Comment:
Urgh, I don't like our standardized title capitalization requiring changing how code is formatted. Can we keep `get-function` lowercase and put it in double backticks, or do we not have styles for that in headings?